### PR TITLE
Grudge

### PR DESCRIPTION
**Pull request by rhardih**
This pull request introduces the "Grudge" feature to ekill as a solution for #9.

It's a continuation of the work started by @duart38, where ekill *remembers* killed elements and tries to kill them on subsequent visits to the same pages.

**Todo:**

- [x] When killing a parent of a previously killed element, replace previous selector with the new one
- [x] Further validation and testing of the MutationObserver approach, as it seems to break on certain pages
- [x] Update Readme
- [x] General cleanup and streamlining; Function, variable and file renaming/moving
- [x] Test / validate on Firefox

### CODE REVIEW

**Comment by rhardih** [comment]
Hey @duart38, it seems you haven't correctly set up git for proper attribution. Currently your commits are authored by `duartsnel@Duarts-MacBook-Pro.local`.

It is my intention to squash and merge this branch when it's ready, which will weed out the commit noise, but if you would like to preserve attribution history, I need to set a proper email as `Co-authored-by` in the final commit. You can read more here: https://help.github.com/articles/about-commit-email-addresses

Let me know.

**Comment by rhardih** [comment]
@duart38 I can see you've set `mr.codercuracao@gmail.com`, as commit author on your other pull request. If it's ok with you, I'll just use that one. Attribution should work just fine with that.

**Comment by rhardih** [comment]
CC @samuel-crypto, @duart38

Still some things to iron out, but the functionality should be in place now.

If you wouldn't mind giving it a test drive, that would be great.

I've found some cases where stuff breaks using the MutationObserver, so that approach might need to backtrack again. I will spend some more time experimenting with it.

**Comment by duart38** [comment]
@rhardih nice ! i will be testing it out

**Comment by rhardih** [comment]
Though it's not perfect. I think the MutationObserver approach is OK for now, given the clearly expressed 'experimental' state of the feature.

File: manifest.json [hunk 1]
```diff
@@ -19,31 +19,37 @@
...
+    "gecko": {
+      "id": "git@ekill.com",
+      "strict_min_version": "42.0"
+    }
+  },
   "permissions": [
-    "activeTab"
+    "activeTab",
+    "storage",
+    "unlimitedStorage"
```
Comment (rhardih) [inline manifest.json]: Hey @duart38,

This slipped my attention on initial review. Is there any specific reason we need this? It seems to only relate to "Web SQL Database and application cache" according to this: https://developer.chrome.com/apps/declare_permissions

Comment (duart38) [reply manifest.json]: i'm not sure, i also got a bit confused by the docs here. i read in some places that the extension or app is limited to 5 MB of local storage, hence why it's there.

Comment (rhardih) [reply manifest.json]: Ok, yeah I see it mentioned here in https://developer.chrome.com/apps/storage#property-local as well.

I doubt anyone will hit 5MB, but I'll just leave it as is then.

**Pull request merged**